---
title: "Process RNAseq from FireCloud"
author: "Guillaume Kugener"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message=FALSE, warning=FALSE)
source('~/Documents/Analysis/RScripts/load_libraries_and_annotations.R')
```

## Overview

This document outlines how we process the RNAseq data from FireCloud in preparation for the DepMap release. The steps are:

* Download the rsem_genes_tpm.txt.gz and rsem_transcripts_tpm.txt.gz files from FireCloud (https://portal.firecloud.org/#workspaces/broad-firecloud-ccle/DepMap_hg38_RNAseq)
* Fix the column names (they are prefixed with ccle2_, ibm_, dm_ and then ARXSPAN_ID. We remove the prefix)
* Compare to the previous release (correlations)
* Quality control: look at the number of genes that have 0 TPM values (proxy for low read count)
* Remove failed samples and save counts, transcripts, and gene files to be uploaded to taiga

```{r include=FALSE}
release <- '19Q2'
directory_to_save_files_to <- '/Volumes/xchip_datasci/gkugener/DepMap_Omics/RNAseq/Expression/data/'

# Make directory for release if it does not exist
dir.create(paste0(directory_to_save_files_to), showWarnings = T)
dir.create(paste0(directory_to_save_files_to, release), showWarnings = T)

# These files are downloaded from FireCloud/Terra
download_paths <- list(
  tpm_genes='~/Downloads/DM19Q2_PATHS_CORRECTED_V2.rsem_genes_tpm.txt.gz',
  tpm_transcripts='~/Downloads/DM19Q2_PATHS_CORRECTED_V2.rsem_transcripts_tpm.txt.gz',
  counts_gene='~/Downloads/DM19Q2_PATHS_CORRECTED_V2.rsem_genes_expected_count.txt.gz',
  exons='~/Downloads/DM19Q2_PATHS_CORRECTED_V2.mergedExonUsage.RData'
)

previous_release_tpm <- load.from.taiga(data.name='depmap-rnaseq-expression-data-363a',
  data.version=15, 
  data.file='CCLE_depMap_19Q1_TPM') %>% 
  set_colnames(stringr::str_extract(string=colnames(.), pattern='ENSG[0-9]+'))
```

Release: `r release`

```{r}

```

```{r}

```

```{r}

```

```{r eval=FALSE}

```

```{r}

```

```{r}

```

```{r}
# Plot the diagonal to see same sample correlations (which should be at or near 1)
# diag(correlation_rnaseq_data_releases)[names(head(number_of_zeros))]
hist(diag(correlation_rnaseq_data_releases), 100)
```

```{r}
# Quick QC
# We are looking for samples with a worrying amount of zeros
zero_threshold <- 39000

number_zeros <- apply(tpm_genes[,3:ncol(tpm_genes)], 2, FUN = function(x) length(x[x == 0]))
nzdf <- data.frame(CL=names(number_zeros), nz=number_zeros, stringsAsFactors = F)
```

```{r}
# Plot of the samples that fail
ggplot(nzdf, aes(nz)) +
  geom_histogram(bins = 100, color='black', fill='white') +
  geom_vline(xintercept = zero_threshold, linetype=2) +
  geom_label_repel(data = nzdf %>% filter(nz > zero_threshold), aes(x=nz, y=0, label=arxspan.to.ccle(CL)), size=2, fill=rgb(1,1,1,0.5))
```

```{r}
pass <- number_zeros[order(-number_zeros)] %>% .[. < zero_threshold] %>% names()

# These samples failed
setdiff(colnames(tpm_genes), pass) %>% .[!(. %in% c('gene_id', 'transcript_id(s)'))]

counts_genes %<>% dplyr::select(c("gene_id","transcript_id(s)", pass))
tpm_genes %<>% dplyr::select(c("gene_id","transcript_id(s)", pass))
transcripts %<>% dplyr::select("transcript_id", "gene_id", pass)
```

```{r eval=TRUE}

```












